### PR TITLE
Refactor Columns naming logic.

### PR DESCRIPTION
**Pull request by irinakhismatullina**
Change columns' names interchange logic.

Also there is some very small refactoring present. Sorry, @vmarkovtsev, in the end it was difficult to detach one from the other.

### CODE REVIEW

File: lookout/style/typos/analyzer.py [hunk 1]
```diff
@@ -152,9 +154,9 @@ def filter_suggestions(self, test_df: pandas.DataFrame,
         Filter suggestions based on the repo specifics and confidence threshold.
 
         :param test_df: DataFrame with info about tested tokens.
-        :param suggestions: Dictionary of correction suggestions grouped by \
+        :param suggestions: Dictionary of correction suggestions grouped by
```
Comment (vmarkovtsev) [inline lookout/style/typos/analyzer.py]: Changes in this file are not related to Columns

File: lookout/style/typos/corrector.py [hunk 2]
```diff
@@ -49,10 +49,15 @@ def __init__(self, **kwargs):
         self.ranker = CandidatesRanker()
 
     @property
-    def threads_number(self):
-        """Return the number of threads used to train and to predict."""
+    def threads_number(self) -> int:
+        """Return the number of threads for multiprocessing used to train and to predict."""
```
Comment (vmarkovtsev) [inline lookout/style/typos/corrector.py]: Here too

File: lookout/style/typos/corrector.py [hunk 3]
```diff
@@ -49,10 +49,15 @@ def __init__(self, **kwargs):
...
         return self.ranker.boost_param["nthread"]
 
+    @threads_number.setter
```
Comment (vmarkovtsev) [inline lookout/style/typos/corrector.py]: And here

File: lookout/style/typos/corrector.py [hunk 4]
```diff
@@ -77,118 +82,88 @@ def initialize_generator(self, vocabulary_file: str, frequencies_file: str,
...
               save_candidates_file: str = None) -> None:
         """
-        Train corrector on the given dataset of typos inside identifiers.
+        Train corrector on the given dataset of typoed tokens inside identifiers.
 
-        :param typos: DataFrame containing columns "typo" and "identifier",
-                      column "token_split" is optional, but used when present.
-        :param candidates: DataFrame with precalculated candidates.
-        :param save_candidates_file: Path to file where to save the candidates.
+        :param data: DataFrame or its .csv dump, containing columns Columns.Token and \
```
Comment (vmarkovtsev) [inline lookout/style/typos/corrector.py]: And all over here

File: lookout/style/typos/corrector.py [hunk 5]
```diff
@@ -77,118 +82,88 @@ def initialize_generator(self, vocabulary_file: str, frequencies_file: str,
...
+        :param return_all: False to return suggestions only for corrected tokens.
+        :param save_candidates_file: Path to file to save candidates to (.pkl).
         :return: Dictionary {id : [[candidate, correctness_proba]]}, candidates are sorted \
                  by correctness probability in a descending order.
         """
-        if candidates is None:
-            candidates = self.generator.generate_candidates(typos, self.threads_number,
-                                                            save_candidates_file)
+        if isinstance(data, str):
+            data = pandas.read_csv(data, index_col=0)
```
Comment (vmarkovtsev) [inline lookout/style/typos/corrector.py]: Etc.

**Comment by irinakhismatullina** [comment]
Done, the PR is split into 4 parts.
